x-pack/test/api_integration/apis/console/autocomplete_entities.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../../ftr_provider_context';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertest');
+
+  describe('/api/console/autocomplete_entities', () => {
+    it('should return an object with properties of "mappings", "aliases", "dataStreams", "legacyTemplates", "indexTemplates", "componentTemplates"', async () => {
+      const expectedFields = [
+        'mappings',
+        'aliases',
+        'dataStreams',
+        'legacyTemplates',
+        'indexTemplates',
+        'componentTemplates',
+      ];
+      const { body } = await supertest
+        .get('/api/console/autocomplete_entities')
+        .query({
+          indices: true,
+          fields: true,
+          templates: true,
+          dataStreams: true,
+        })
+        .expect(200);
+      expect(Object.keys(body).sort()).to.eql(expectedFields.sort());
+    });
+
+    describe('with template settings set to false', () => {
```

Comment:
Can we change this to test behavior when all settings are set to false? Then we can add more assertions about the other properties of the response payload to more thoroughly cover this API's behavior.